solutions/src/main/java/ru/sberbank/school/task11/MyStreams.java

```diff
@@ -0,0 +1,159 @@
+package ru.sberbank.school.task11;
+
+import lombok.NonNull;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+/**
+ * Облегченная версия класса. {@link Stream}
+ */
+public class MyStreams<T> {
+
+    private List<T> elements;
+
+    @SafeVarargs
+    private MyStreams(T...elements) {
+        this.elements = Arrays.asList(elements);
+    }
+
+    private MyStreams(Collection<T> elements) {
+        this.elements = new ArrayList<>(elements);
+    }
+
+    /**
+     * Принимает на вход массив элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    @SafeVarargs
+    public static <T> MyStreams<T> of(@NonNull T... elements) {
+        return new MyStreams<>(elements);
+    }
+
+    /**
+     * Принимает на вход коллекцию элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    public static <T> MyStreams<T> of(@NonNull Collection<T> elements) {
+        return new MyStreams<>(elements);
+    }
+
+    /**
+     * Фильтрация элементов по заданному правилу. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     *
+     * <p>Иными словами, этот метод оставляет в коллекции только те элементы, которые удовлетворяют условию в лямбде.
+     *
+     * <p>Аналог Stream#map(Function).
+     *
+     * <p>Intermediate операция.
+     *
+     * @param predicate - правило фильтрации элементов.
+     * @return стрим
+     */
+    public MyStreams<T> filter(@NonNull Predicate<? super T> predicate) {
+        List<T> list = new ArrayList<>();
+
+        for (T element : elements) {
+            if (predicate.test(element)) {
+                list.add(element);
+            }
+        }
+
+        return new MyStreams<>(list);
+    }
+
+    /**
+     * Преобразование элементов в какие-то другие элементы. Необходимо подобрать нужый интерфейс для передачи в этот метод.
```

Comment:
Line is longer than 120 characters (found 123).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-862413266>*